meta-client/src/test/java/org/apache/helix/metaclient/puppy/AbstractPuppy.java

```diff
@@ -55,30 +55,23 @@ public AbstractPuppy(MetaClientInterface<String> metaclient, PuppySpec puppySpec
   @Override
   public void run() {
     try {
-      while (true) {
+      while (!Thread.currentThread().isInterrupted()) {
```

Comment:
Was this main thread set daemon? If yes, we don't have check the interrupted here.

Reply:
No the thread is not a daemon thread.